Sources/Subjects/ReplaySubject.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  ReplaySubject.swift
+//  CombineExt
+//
+//  Created by Jasdev Singh on 13/04/2020.
+//  Copyright © 2020 Combine Community. All rights reserved.
+//
+
+import Combine
+
+/// A `ReplaySubject` is a subject that can buffer one or more values. It stores value events, up to its `bufferSize` in a
+/// first-in-first-out manner and then replays it to
+/// future subscribers and also forwards completion events.
+///
+/// The implementation borrows heavily from [Entwine’s](https://github.com/tcldr/Entwine/blob/b839c9fcc7466878d6a823677ce608da998b95b9/Sources/Entwine/Operators/ReplaySubject.swift).
+public final class ReplaySubject<Output, Failure: Error>: Subject {
+    public typealias Output = Output
+    public typealias Failure = Failure
+
+    private let bufferSize: Int
+    private var buffer = [Output]()
+
+    // Keeping track of all live subscriptions, so `send` events can be forwarded to them.
+    private var subscriptions = [Subscription<AnySubscriber<Output, Failure>>]()
+
+    // We also track subscriber identifiers, to more quickly bottom-out double subscribes instead of having to do a
+    // linear pass over `subscriptions`.
+    private var subscriberIdentifiers = Set<CombineIdentifier>()
+
+    private var completion: Subscribers.Completion<Failure>?
+    private var isActive: Bool { completion == nil }
+
+    /// Create a `ReplaySubject`, buffering up to `bufferSize` values and replaying them to new subscribers
+    /// - Parameter bufferSize: The maximum number of value events to buffer and replay to all future subscribers.
+    public init(bufferSize: Int) {
+        self.bufferSize = bufferSize
+    }
+
+    public func send(_ value: Output) {
+        guard isActive else { return }
+
+        buffer.append(value)
+
+        if buffer.count > bufferSize {
+            buffer.removeFirst()
+        }
+
+        subscriptions.forEach { $0.forwardValueToBuffer(value) }
+    }
+
+    public func send(completion: Subscribers.Completion<Failure>) {
+        guard isActive else { return }
+
+        self.completion = completion
+
+        subscriptions.forEach { $0.forwardCompletionToBuffer(completion) }
+    }
+
+    public func send(subscription: Combine.Subscription) {
+        subscription.request(.unlimited)
+    }
+
+    public func receive<Subscriber: Combine.Subscriber>(subscriber: Subscriber) where Failure == Subscriber.Failure, Output == Subscriber.Input {
+        let subscriberIdentifier = subscriber.combineIdentifier
+
+        guard !subscriberIdentifiers.contains(subscriberIdentifier) else {
+            subscriber.receive(subscription: Subscriptions.empty)
+            return
+        }
+
+        let subscription = Subscription(downstream:
+        AnySubscriber(subscriber)) { [weak self] in
```

Comment:
Sorry for the nitpick, I'll just commit this in
```suggestion
        let subscription = Subscription(downstream: AnySubscriber(subscriber)) { [weak self] in
```